internal/daysteps/daysteps.go

```diff
@@ -1,13 +1,99 @@
 package daysteps
 
+import (
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Yandex-Practicum/tracker/internal/personaldata"
+	"github.com/Yandex-Practicum/tracker/internal/spentenergy"
+)
+
 type DaySteps struct {
-	// TODO: добавить поля
+	Steps    int
+	Duration time.Duration
+	personaldata.Personal
 }
 
 func (ds *DaySteps) Parse(datastring string) (err error) {
-	// TODO: реализовать функцию
+	parts := strings.Split(datastring, ",")
+	if len(parts) != 2 {
+		return errors.New("неверный формат данных: ожидается 2 части")
+	}
+
+	stepsStr := parts[0]
+	if stepsStr == "" {
+		return errors.New("количество шагов не может быть пустым")
+	}
+
+	if strings.TrimSpace(stepsStr) != stepsStr {
+		return errors.New("неверный формат шагов: пробелы в начале или конце не допускаются")
+	}
+
+	hasNonDigit := false
+	for i, char := range stepsStr {
+		if char == '+' || char == '-' {
+
+			if i != 0 {
+				hasNonDigit = true
+				break
+			}
+		} else if char < '0' || char > '9' {
+			hasNonDigit = true
+			break
+		}
+	}
+
+	if hasNonDigit {
+		return errors.New("неверный формат шагов: должны быть только цифры, возможен знак в начале")
+	}
+
+	steps, err := strconv.Atoi(stepsStr)
+	if err != nil {
+		return fmt.Errorf("ошибка парсинга шагов: %v", err)
```

Comment:
При оборачивании нужно использовать `%w` (wrap), а просто при выводе в консоль `%v`
```suggestion
		return fmt.Errorf("ошибка парсинга шагов: %w", err)
```